Assets/MixedRealityToolkit/_Core/Devices/MixedReality/WindowsMixedRealityDeviceManager.cs

```diff
@@ -115,12 +110,20 @@ private WindowsMixedRealityController GetOrAddController(InteractionSourceState
                     throw new ArgumentOutOfRangeException();
             }
 
-            var inputSource = InputSystem?.RequestNewGenericInputSource($"Mixed Reality Controller {controllingHand}");
+            IMixedRealityPointer[] pointers = interactionSourceState.source.supportsPointing ? RequestPointers(typeof(WindowsMixedRealityController), controllingHand) : null;
+            var inputSource = InputSystem?.RequestNewGenericInputSource($"Mixed Reality Controller {controllingHand}", pointers);
             var detectedController = new WindowsMixedRealityController(TrackingState.NotTracked, controllingHand, inputSource);
+
             detectedController.SetupConfiguration(typeof(WindowsMixedRealityController));
-            detectedController.UpdateController(interactionSourceState);
```

Comment:
Why don't we update the controller data here anymore? Now the data in a `SourceDetected` event will be empty, or is it properly filled in elsewhere?

Reply:
Update is already called in the GetOrAddCotroller, so this was in effect calling it twice.

Reply:
This _is_ `GetOrAddController`

Reply:
Haha, I think he meant after we call GetOrAddCotroller we usually update it too.

We were running into issues, and this fixes it.
Essentially we were updating the controller before we raised the source found.

Reply:
Shouldn't be raising input changes, no.
Should be filling in the state though, in case a dev wants to query it in `SourceDetected`. Is that happening?

Reply:
Yes. after we get the controller.

Reply:
In here? https://github.com/StephenHodgson/MixedRealityToolkit-Unity/blob/c2e196261d24ed8774589ba3b073b963e4ab2388/Assets/MixedRealityToolkit/_Core/Devices/MixedReality/WindowsMixedRealityDeviceManager.cs#L149

Reply:
Let me add an update there and test it out to see if we're still getting that race condition.

Reply:
Doesn't seem to have the race condition, so I added the update after raise detected